app/Http/Controllers/ReservationController.php

```diff
@@ -111,32 +112,70 @@ public function index()
     public function store(Request $request)
     {
 
-        //   $request->validate([
-        //      'name' => 'required'
-        // ]);
+        $request->validate([
+            'name' => 'required',
+            'email' => 'required',
+            'tel' => 'required',
+            'consent' => 'required'
+        ]);
+        $seatsData =  json_decode($request->input('seats'), true);
+
+        if ($request->input('stand') == 0 &&  count($seatsData ?? []) == 0) {
+            return response()->json([
+                'message' => 'Either seats or stand tickets must be filled!'
+            ], 400);
+        }
+
         $stand = $request->input('stand') ?? 0;
-        $price = 500;
+
+        $seats = Seat::findMany($seatsData)->toArray();
+
+        $totalPrice = $this->getStandPrice($stand) + $this->getSeatsPrice($seats);
+
         $reservation = Reservation::create(
             [
                 'name' =>  $request->input('name'),
                 'email' => $request->input('email'),
                 'tel' => $request->input('tel'),
                 'note' => $request->input('note'),
                 'stand' => $stand,
-                'price_all' => $stand * $price,
+                'price_all' => $totalPrice,
                 'status' => 1,
-                //'date_reservation' => Carbon::now(),
-                'date_payment' => Carbon::now()
+                'consent' => (int)$request->input('consent'),
+                'date_payment' => null
             ]
         );
-        $seats = Seat::findMany($request->input('seats'))->toArray();
+
+
         $this->updateSeats($seats, $reservation->id);
 
         $data = ['reservation' => $reservation, 'seats' => $seats];
 
-        // $this->sendEmail();
+        return response()->json($data, 200);
+    }
+
+    private function getStandPrice($stand)
+    {
+        $priceStand = 350;
+        return $stand * $priceStand;
+    }
+    private function getSeatsPrice($seats)
+    {
+
+        $priceSit = 500;
+        $priceSitRaut = 750;
+
+        $totalPrice = 0;
+        foreach ($seats as $seat) {
+            if ($seat['typ'] == "raut") {
```

Comment:
is there a reason to use double quotes for `"raut"` ?

Reply:
no reason at all, replaced with singe quotes